### DISCUSSION TITLE
Appreciate this project and have some thoughts

### DISCUSSION THREAD

**Discussion by oldgristmill** [mcguirepr89/BirdNET-Pi#169] [Q&A]
After playing with the BirdNET app on my phone, I'm really excited to play around with this software once my Pi arrives later this week. It's already got me thinking about how I might use it going forward... so I have a couple thoughts/questions:

1) I run a heritage site in a rural area with a large riparian area and lots of gardens, so anticipate plenty of IDs once I've got the system mounted and running. Eventually, I hope to take this data and display some sort of stripped down real-time dashboard for the public on a display elsewhere on site and maybe even on our own website, rather than giving direct public access to a Pi sitting on our network. Looking forward to figuring out how that might work, eventually.

2) Has anyone thought through using this stack of software with a different TFlite model?  If there was a pretrained model for, say, bat or insect sound identification, would this collection of software generalize well enough to make that a feasible fork project?

I'm an end-user, not a developer, so have to rely on the amazing work you smart people are doing (which is deeply appreciated!) so my dreams and ideas are always going to be larger than my capacity to create the solutions myself, heh. (Probably the worst kind of person to have offering comment on a github repository...)

**Comment by mcguirepr89** [comment]
Hi, @oldgristmill -- Thanks for starting this discussion!

Thank you also for your interest in BirdNET-Pi and for your great thoughts -- it definitely seems that once you get started playing around with the system that you'll know exactly how it might fit in at the heritage site. I'm hopeful, too, that once you get a change to play around with your system and discover the various ways that are available to keep it secure, you may be interested in allowing your visitors full access to the web interface. Your discretion, of course, is to be used, and I would love to address your questions to help you figure out the best way to proceed:
Note: My responses may seem to contradict what is written in the `main` branch's README.md, but these answers anticipate the changes that will be apart of the `main` branch soon after I am able to verify through testing that the `newif` branch is meeting all user needs.

>1. . . . Eventually, I hope to take this data and display some sort of stripped down real-time dashboard for the public on a display elsewhere on site and maybe even on our own website . . .

The data is all available in a small SQLite database, though that means the data is a file on the BirdNET-Pi and not a database server. I know you're not a developer, but there are several ways one might work around this to achieve remote access to the information you desire:
1. Convert the installation to using MariaDB instead of SQLite, so that the system can serve the database via a port (or a socket that is handled with a reverse_proxy) where remote computers can fetch the data.
2. Use the SSH server and a cron job to create a backup of the full SQLite database on a remote machine so that the remote machine can address the database locally.
3. Make the database available in some sort of network-attached storage, (NAS, SSHFS, Samba Share, HTTPS web server), all which can be served from the BirdNET-Pi itself if needed.
4. Create regular reports for the data you wish to display and upload that data to the remote source using whatever protocol the remote computer accepts (SFTP, PHP uploads via API, HTTP POSTs, eMail, etc)

The "real-time" aspect of the reporting would require you utilize the first option, while everything else can be automated with time intervals, e.g., updated every 5/15/60/howevermany minutes.

> . . . rather than giving direct public access to a Pi sitting on our network.

If I may make a note here: unless the public has unprotected access to your Local Area Network (LAN), that is to say, can connect to your WiFi without having to be given the password, the public would not have any access to your installation. You can certainly make your installation public, and there are settings within BirdNET-Pi that will help in securing that public access (Forced HTTPS and Basic-Authentication over HTTPS [which is bank-level security] to protect all configuration settings, the live audio stream, and the little bit of raw data that is temporarily saved). But by default, if the installation is performed within your LAN, it will stay in your LAN.
Likewise, you would need to forward the appropriate ports to allow the public's HTTP requests to be answered by the BirdNET-Pi.

Lastly, I'm happy to help you figure out any other places you may want to "lock down" to feel more comfortable allowing folks (including possibly nefarious strangers) access to your BirdNET-Pi installation.

>2. Has anyone thought through using this stack of software with a different TFlite model? If there were a pretrained model for, say, bat or insect sound identification, would this collection of software generalize well enough to make that a feasible fork project?

I have wondered this same thing, and my supposition is YES, because aside from the analysis server loading the specific BirdNET model, the rest of the software just reads from the resulting .csv file that BirdNET creates. I say that as someone who has had NOTHING to do with building nor training the model, though I would be more than happy (giddy) to take a whack at creating a BugNET-Pi or a FrogNET-Pi, or BatNET-Pi, or WhaleNET-Pi, or AmbulanceNET-Pi, or GunshotNET-Pi, or anything else smart people can come up with!

I personally want to see a BugNET and BatNET model to build around -- _that_ would be super cool! Some folks have used the BirdNET-Pi along with bat-frequency recording equipment to monitor the bats in real-time with the spectrograms and live-audio stream, though no audio extractions are made since there is no model to trigger the audio extraction script.

Wonderful ideas and very valuable feedback! Please keep them coming and let me know what other ideas and questions come up.

Please accept my best regards and sincere hopes that all goes well during your first installation,

Patrick

**Comment by oldgristmill** [comment]
Thanks for your thoughts! My Pi arrived the other night and I got it set up and working (in the wee hours, so was using youtube videos of bird calls to test it, heh). Secured a slightly better microphone, which is working for now. In the immediate, I was more interested in IDs than quality recordings, but the pull is strong...

I've written for a BirdWeather ID and hope to make this publicly accessible soon. Thanks again for all your hard work!

**Reply by mcguirepr89** [reply]
The temptation to improve the recording equipment is real!! and indeed it grows!!! and it's fun!!

For what it is worth, [I can recommend this HIGHLY](https://github.com/mcguirepr89/BirdNET-Pi/discussions/39#discussioncomment-2180372) -- I use that hardware (thanks to @DD4WH Frank!) and can attest that it makes for great recordings!
